opentelemetry-otlp/tests/integration_test/tests/metrics.rs

```diff
@@ -30,7 +30,7 @@ async fn init_metrics() -> SdkMeterProvider {
     let exporter = create_exporter();
 
     let reader = PeriodicReader::builder(exporter)
-        .with_interval(Duration::from_millis(100))
+        .with_interval(Duration::from_millis(500))
```

Comment:
the sleep is around 5 seconds, so 500 msec interval is fine, and this reduces the amount of internal logs.
Also, I am thinking if we should rely on force_flush/shutdown for integration tests more, to avoid this sleep requirement.  We should still have one set of tests for testing the export triggered by interval.

Reply:
I think we'll still need a bit of a buffer because of the flushing on the otlp-collector side? What I think what would be _the nicest_ would be to setup a simple, short exponential backoff mechanism in here - the function knows what it is looking for for the particular test, so it's well positioned to decide to wait a bit and look again: 

https://github.com/open-telemetry/opentelemetry-rust/blob/9011f63ff653e1e1cde5e17cbcf3cdb71ab280ce/opentelemetry-otlp/tests/integration_test/tests/metrics.rs#L80

so that we can use a tighter timing in the best case.

I've also added this `rotation` thing for logs, [which I believe](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18251#issuecomment-1414497442) will decrease buffering on the collector side - doesn't help us with the other signals, though:

https://github.com/open-telemetry/opentelemetry-rust/blob/9011f63ff653e1e1cde5e17cbcf3cdb71ab280ce/opentelemetry-otlp/tests/integration_test/otel-collector-config.yaml#L12-L14